### PR TITLE
Feature/validations

### PR DESCRIPTION
**Pull request by Thi-Ponce**
## Learning objectives
- Use validations for models.
- Write unit tests for models.
- Secure app from n+1 problems.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -13,17 +13,19 @@ def show
...
     @post = Post.new
-    @current_user_id = current_user.id.to_i
+    @user = User.find(params[:user_id])
   end
 
   def create
     @user = User.find(params[:user_id])
 
-    @post = current_user.posts.new(params.permit(:title, :text))
+    @post = current_user.posts.new(params.require(:post).permit(:title, :text))
```
Comment (rustam-tolipov) [inline app/controllers/posts_controller.rb]: - [x] Kindly move permitted params after methods.


example: 
```
private

  def comment_params
    params.require(:comment).permit(:text)
  end
```

File: spec/requests/posts_controller_spec.rb [hunk 2]
```diff
@@ -1,25 +1,29 @@
-require 'rails_helper'
+# require 'rails_helper'
 
-RSpec.describe 'Posts requests', type: :request do
-  describe 'tests for index posts' do
-    before(:example) { get user_posts_path(user_id: 1) }
+# RSpec.describe 'Posts requests', type: :request do
```
Comment (rustam-tolipov) [inline spec/requests/posts_controller_spec.rb]: - [x] Please remove unused commented codes from your file!

File: spec/requests/users_controller_spec.rb [hunk 3]
```diff
@@ -1,33 +1,33 @@
-require 'rails_helper'
+# require 'rails_helper'
 
-RSpec.describe 'UsersControllers', type: :request do
-  it 'Renders users and give out the correct status' do
-    get users_path
-    expect(response).to have_http_status(:ok)
-  end
+# RSpec.describe 'UsersControllers', type: :request do
```
Comment (rustam-tolipov) [inline spec/requests/users_controller_spec.rb]: - [x] Please remove unused commented codes from your file!

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+RSpec.describe Comment, type: :model do
+  subject do
+    user = User.create(name: 'Graha Tia', photo: '', bio: 'Crystal exharc')
+    post = Post.new(author: user, title: 'Stand tall my friends', text: 'Our journey will never end')
+    Comment.new(author: user, post:, text: 'May those who walked before, lead those who walk after')
+  end
+
+  before { subject.save }
+
```
Comment (rustam-tolipov) [inline spec/models/comment_spec.rb]: - [x] You have some errors on your test files, please fix them for the next step.


>   /Users/rustamjon/Rustamxon7/blog-appssss/spec/models/comment_spec.rb:7: syntax error, unexpected ','
>   ...omment.new(author: user, post:, text: 'May those who walked ...

File: spec/models/like_spec.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+RSpec.describe Like, type: :model do
+  subject do
+    user = User.create(name: 'Haurchefant', photo: '', bio: 'In loving memory of our dear partner')
+    post = Post.new(author: user, title: 'Do not look at me so', text: 'A smile better suits a hero')
+    Like.new(author: user, post:)
+  end
+
+  before { subject.save }
+
```
Comment (rustam-tolipov) [inline spec/models/like_spec.rb:11]: - [x] Your specs are showing errors, kindly fix them for the next step.


SyntaxError:
  /Users/rustamjon/Rustamxon7/blog-appssss/spec/models/like_spec.rb:7: syntax error, unexpected ')'
  ...  Like.new(author: user, post:)
  ...                              ^

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @Thi-Ponce, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_

emojibest_com_AnimatedSticker


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/like_spec.rb (hunk 5 above)
Comment (Thi-Ponce) [reply spec/models/like_spec.rb:11]: Hello @Rustamxon7 , thanks for your review.

I ran the tests and the linters and I did not get any of those errors, would you mind explaining how you got them? 
pic 4

**Review by keneogu — CHANGES_REQUESTED** [review]
> ## Hi @Thi-Ponce,
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
> 
> ## Congratulations!     
> Cheers and Happy coding!clapclapclap
> 
> image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966) image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966)
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**